src/MetadataUtility/Metadata/FrontierLabs/FlacCommentExtractor.cs

```diff
@@ -27,7 +29,71 @@ public ValueTask<bool> CanProcessAsync(TargetInformation information)
+                        {
+                            ManufacturerID = recording.MemoryCard?.ManufacturerID ?? (byte)((Dictionary<string, object>)parsedComments[FrontierLabs.SdCidCommentKey])[SdCardCid.ManufacturerIDKey],
+                            OEMID = recording.MemoryCard?.OEMID ?? (string)((Dictionary<string, object>)parsedComments[FrontierLabs.SdCidCommentKey])[SdCardCid.OEMIDKey],
+                            ProductName = recording.MemoryCard?.ProductName ?? (string)((Dictionary<string, object>)parsedComments[FrontierLabs.SdCidCommentKey])[SdCardCid.ProductNameKey],
+                            ProductRevision = recording.MemoryCard?.ProductRevision ?? (float)((Dictionary<string, object>)parsedComments[FrontierLabs.SdCidCommentKey])[SdCardCid.ProductRevisionKey],
+                            SerialNumber = recording.MemoryCard?.SerialNumber ?? (uint)((Dictionary<string, object>)parsedComments[FrontierLabs.SdCidCommentKey])[SdCardCid.SerialNumberKey],
+                            ManufactureDate = recording.MemoryCard?.ManufactureDate ?? (string)((Dictionary<string, object>)parsedComments[FrontierLabs.SdCidCommentKey])[SdCardCid.ManufactureDateKey],
+                        },
+                    };
+
+                    foreach (Dictionary<string, object> newMicrophone in ((Dictionary<string, Dictionary<string, object>>)parsedComments[FrontierLabs.MicrophonesKey]).Values)
+                    {
+                        for (int i = 0; i < 2; i++)
+                        {
+                            if (recording.Sensor.Microphones[i] == null || recording.Sensor.Microphones[i].UID.Equals((string)newMicrophone[FrontierLabs.MicrophoneUIDCommentKey]))
+                            {
+                                if (!((string)newMicrophone[FrontierLabs.MicrophoneUIDCommentKey]).Equals(FrontierLabs.UnknownMicrophoneString))
+                                {
+                                    recording.Sensor.Microphones[i] = (recording.Sensor.Microphones[i] ?? new Microphone()) with
+                                    {
+                                        Type = recording.Sensor.Microphones[i]?.Type ?? (string)newMicrophone[FrontierLabs.MicrophoneTypeCommentKey],
+                                        UID = recording.Sensor.Microphones[i]?.UID ?? (string)newMicrophone[FrontierLabs.MicrophoneUIDCommentKey],
+                                        BuildDate = recording.Sensor.Microphones[i]?.BuildDate ?? (string)newMicrophone[FrontierLabs.MicrophoneBuildDateCommentKey],
+                                        Gain = recording.Sensor.Microphones[i]?.Gain ?? (string)newMicrophone[FrontierLabs.MicrophoneGainCommentKey],
+                                    };
+                                }
+
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
OK this is not leaky (good) but it's a right mess (not your fault).

I would suggest breaking this up a bit by pulling common expressions our into variables.